tutorials/2022-02-09-neow3j-nep11/index.md

```diff
@@ -0,0 +1,647 @@
+
+The method `burn` can be invoked by the owner of a token. It deletes all information about the token and updates the balance and total supply accordingly. If the intent of burning a token need not require the storage to be freed, the token could also just be sent to a *burner address*.
+
+```java
+    @Safe
+    public static Hash160 contractOwner() {
+        return contractOwner;
+    }
+
+    public static void mint(Hash160 owner, ByteString tokenId, Map<String, String> properties) {
+        assert Runtime.checkWitness(contractOwner) : "No authorization.";
+        assert registryMap.get(tokenId) == null : "This token id already exists.";
+        assert properties.containsKey(propName) : "The properties must contain a value for the key 'name'.";
+
+        String tokenName = properties.get(propName);
+        propertiesNameMap.put(tokenId, tokenName);
+
+        if (properties.containsKey(propDescription)) {
+            String description = properties.get(propDescription);
+            propertiesDescriptionMap.put(tokenId, description);
+        }
+        if (properties.containsKey(propImage)) {
+            String image = properties.get(propImage);
+            propertiesImageMap.put(tokenId, image);
+        }
+        if (properties.containsKey(propTokenURI)) {
+            String tokenURI = properties.get(propTokenURI);
+            propertiesTokenURIMap.put(tokenId, tokenURI);
+        }
+
+        registryMap.put(tokenId, tokenId);
+        ownerOfMap.put(tokenId, owner.toByteArray());
+        new StorageMap(ctx, createTokensOfPrefix(owner)).put(tokenId, 1);
+
+        increaseBalanceByOne(owner);
+        incrementTotalSupplyByOne();
+        onMint.fire(owner, tokenId, properties);
+    }
+
+    public static boolean burn(ByteString tokenId) {
+        Hash160 owner = ownerOf(tokenId);
+        assert owner != null : "This token id does not exist.";
+        throwIfSignerIsNotOwner(owner);
+
+        registryMap.delete(tokenId);
+        propertiesNameMap.delete(tokenId);
+        propertiesDescriptionMap.delete(tokenId);
+        propertiesImageMap.delete(tokenId);
+        propertiesTokenURIMap.delete(tokenId);
+        ownerOfMap.delete(tokenId);
+
+        new StorageMap(ctx, createTokensOfPrefix(owner)).delete(tokenId);
+        decreaseBalanceByOne(owner);
+        decrementTotalSupplyByOne();
+        return true;
+    }
+```
+
+### Private Helper Methods
+
+Private methods can be used to extract code that is used internally in the contract. The following private methods are used in the NEP-17 example contract. For example, in order to prevent writing the same exception message to check the witness of the owner, the private method `throwIfSignerIsNotContractOwner` allows to write this code only once.
```

Comment:
```suggestion
Private methods can be used to, for example, simplify and make the smart contract a bit more readable. The following private methods are used in the NEP-17 example contract. For example, in order to prevent writing the same exception message to check the witness of the owner, the private method `throwIfSignerIsNotContractOwner` allows writing this code only once.
```